workbench-core/datasets/src/dataSetService.ts

```diff
@@ -766,4 +789,31 @@ export class DataSetService {
     );
     return { data: { mountObject } };
   }
+
+  private async _updateNewDataSetPermissions(
+    dataset: DataSet,
+    request: CreateProvisionDatasetRequest
+  ): Promise<DataSetPermission[]> {
+    if (dataset.id) {
```

Comment:
could we early return here?
```
if (!dataset.id) {
  return [];
}
```

Reply:
sure! I was in a group for years that maintained as "one return per method" rule. Can you tell?